src/main/java/cn/nukkit/entity/Entity.java

```diff
@@ -577,6 +577,10 @@ public abstract class Entity extends Location implements Metadatable {
     private int maxHealth = 20;
     private volatile boolean initialized;
 
+    @PowerNukkitXOnly
+    @Since("1.20.0-r2")
+    protected boolean saveWithChunk = true;
```

Comment:
it should be volatile since server may crash and save the chunk while the entity object is constructing.